apis/goharbor.io/v1alpha2/chartmuseum_types.go

```diff
@@ -0,0 +1,356 @@
+package v1alpha2
+
+import (
+	harbormetav1 "github.com/goharbor/harbor-operator/apis/meta/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +k8s:openapi-gen=true
+// +resource:path=chartmuseum
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:categories="goharbor"
+// +kubebuilder:printcolumn:name="Version",type=string,JSONPath=`.spec.version`,description="The semver Harbor version",priority=5
+// +kubebuilder:printcolumn:name="Replicas",type=string,JSONPath=`.spec.replicas`,description="The number of replicas",priority=0
+// ChartMuseum is the Schema for the registries API.
+type ChartMuseum struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ChartMuseumSpec `json:"spec,omitempty"`
+
+	Status harbormetav1.ComponentStatus `json:"status,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// ChartMuseumList contains a list of ChartMuseum.
+type ChartMuseumList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ChartMuseum `json:"items"`
+}
+
+// ChartMuseumSpec defines the desired state of ChartMuseum.
+type ChartMuseumSpec struct {
```

Comment:
The chartmeseum is an underlying compoennt of Harbor. And there is a adapter layer in harbor core for adapting chartmeseum. Not all of the chartmeseum configurations can be exposed and allowed to do modifications.

Reply:
See it more like the configuration of *ChartMuseum* standalone component.
The Harbor controller (`controllers/goharbor/harbor/`) will set the configuration for its own usage, so with the basic auth and the depth to 1.
IMO this way is more flexible for future development and more readable for users.

About this *special* configuration that *must* be set to a specific value (cf <https://github.com/goharbor/harbor-operator/pull/59#discussion_r467072830> and <https://github.com/goharbor/harbor-operator/pull/59#discussion_r467072327>), I will add comments in CRDs to warn about that.
WDYT?